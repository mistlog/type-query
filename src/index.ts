export * from "./type-babel";
export * from "./type-query";
export * from "./type-runtime";