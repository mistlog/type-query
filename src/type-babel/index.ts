export * from "./basic";
export * from "./literal";
export * from "./object";