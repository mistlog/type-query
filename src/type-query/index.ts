export * from "./core";
export * from "./extension";