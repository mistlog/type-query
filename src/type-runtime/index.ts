export * from "./type-runtime";
export * from "./type-context";