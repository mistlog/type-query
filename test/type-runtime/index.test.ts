import path from "path";
import fs from "fs";
import { TypeRuntime } from "../../src";

describe("type runtime", () => {
    test("get type info", () => {
        const filePath = path.resolve(__dirname, "../assets/core-basic.type")
        const input = fs.readFileSync(filePath, "utf-8");

        const runtime = new TypeRuntime(filePath, input);
        const typeInfo = runtime.run();

        expect(typeInfo.tsType).toMatchSnapshot();
    })
})