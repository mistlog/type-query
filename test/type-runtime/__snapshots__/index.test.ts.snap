// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type runtime get type info 1`] = `
Object {
  "Todo": "{ title: string; description: string; completed: boolean; }",
  "TodoPreview": "[\\"use js\\", \\"return $.use(\\\\\\"Todo\\\\\\")\\\\r\\\\n    .type(); \\\\r\\\\n\\"]",
}
`;
